maxim/models/dataset.py

```
@@ -9,128 +10,234 @@ class VariableType(str):
-            DatasetEntry: The created dataset entry
+            List[DatasetEntryWithRowNo]: List of DatasetEntryWithRowNo objects, one for each column
             
         Raises:
-            KeyError: If required input field is missing
-            ValueError: If data format is invalid
+            ValueError: If the dataset entry is invalid
         """
-        if not isinstance(data, dict):
-            raise ValueError("Input data must be a dictionary")
+        if not isinstance(dataset_entry, DatasetEntry):
+            raise TypeError("dataset_entry must be a DatasetEntry")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)